src/backend/gl/gl_common.c

```diff
@@ -1008,23 +1008,66 @@ void *gl_create_blur_context(backend_t *base, enum blur_method method, void *arg
 		// Build shader
 		int width = kern->w, height = kern->h;
 		int nele = width * height;
+		// '%.7g' is at most 14 characters, inserted 3 times
 		size_t body_len = (strlen(shader_add) + 42) * (uint)nele;
```

Comment:
Since one of the for loops will only run for `ceil(width/2)` or `ceil(height/2)` iterations, `body_len` will be too large. Being some bytes to large shouldn't be a problem. Being at least one to short however is bad.
If the buffer should be tighter, this may be computed depending on which loop will run with less iterations.